### PR TITLE
General improvements ported from the hackathon efforts:

### PR DESCRIPTION
**Pull request by tskisner**
* Add "global" operator dependency type for objects in the data
  dictionary.

* Clean up incorrect operator dependencies.

* Add a boolean return to the detdata ensure() method, so that
  operators can tell if the data was already existing and short
  circuit calculations.

### CODE REVIEW

**Review by nestordemeure — APPROVED** [review]
Looking good (liking the fact that it adds code but also comments). Should be okay to merge as soon as it passes tests.

File: src/toast/ops/sim_tod_atm.py [hunk 1]
```diff
@@ -303,13 +303,23 @@ def _exec(self, data, detectors=None, **kwargs):
         loading_key = f"{self.name}_loading"
 
         # Set up the observing operator
+        if self.shared_flags is None:
+            # Cannot observe samples that have no pointing
+            shared_flags = self.detector_pointing.shared_flags
+            shared_flag_mask = self.detector_pointing.shared_flag_mask
+        else:
+            # Trust that the user has provided a flag that excludes smaples
```
Comment (keskitalo) [inline src/toast/ops/sim_tod_atm.py:311]: smaples -> samples

File: src/toast/ops/stokes_weights.py [hunk 2]
```diff
@@ -168,6 +168,23 @@ def _exec(self, data, detectors=None, **kwargs):
                 # Nothing to do for this observation
                 continue
 
+            # Create (or re-use) output data for the weights
+
+            if self.single_precision:
+                existed = ob.detdata.ensure(
```
Comment (keskitalo) [inline src/toast/ops/stokes_weights.py:174]: This PR uses `exists` and `existed` for the return value. Are we happy with that or should we pick one and use it consistently?

File: src/toast/scripts/toast_mini [hunk 3]
```diff
@@ -117,6 +119,10 @@ def parse_arguments():
     args.psd_net = 50.0e-6
     args.psd_fmin = 1.0e-5
 
+    # DEBUGGING
```
Comment (keskitalo) [inline src/toast/scripts/toast_mini:122]: Are these left here intentionally?

**Review by keskitalo — APPROVED** [review]
Go ahead and merge as soon as the tests pass, unless you want to act on the small comments.

File: src/toast/ops/sim_tod_atm.py (hunk 1 above)
Comment (tskisner) [reply src/toast/ops/sim_tod_atm.py]: fixed

File: src/toast/ops/stokes_weights.py (hunk 2 above)
Comment (tskisner) [reply src/toast/ops/stokes_weights.py]: Good point, changed to "exists" everywhere.

**Pull request merged**